File: frontend/src/api/categories.ts
import api from './index'
import type { ApiResponse, Category } from '../types'

export const getCategories = () =>
  api.get<ApiResponse<Category[]>>('/categories')

export const createCategory = (data: { name: string; type: 'income' | 'expense'; icon?: string; sort_order?: number }) =>
  api.post<ApiResponse<Category>>('/categories', data)

export const updateCategory = (id: number, data: { name?: string; icon?: string; sort_order?: number }) =>
  api.put<ApiResponse<Category>>(`/categories/${id}`, data)

export const deleteCategory = (id: number, confirm = false) =>
  api.delete<ApiResponse<null>>(`/categories/${id}?confirm=${confirm}`)

File: backend/app/routers/categories.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session
from typing import Optional
from app.database import get_db
from app.schemas.category import CategoryCreate, CategoryUpdate, CategoryResponse
from app.services import category_service
from app.middleware.auth import get_current_user_id

router = APIRouter(prefix="/api/categories", tags=["categories"])


@router.get("")
def list_categories(user_id: int = Depends(get_current_user_id), db: Session = Depends(get_db)):
    items = category_service.get_categories(db, user_id)
    return {"code": 0, "message": "success", "data": items}


@router.post("")
def create_category(data: CategoryCreate, user_id: int = Depends(get_current_user_id), db: Session = Depends(get_db)):
    item = category_service.create_category(db, user_id, data)
    return {"code": 0, "message": "创建成功", "data": item}


@router.put("/{category_id}")
def update_category(category_id: int, data: CategoryUpdate, user_id: int = Depends(get_current_user_id), db: Session = Depends(get_db)):
    item = category_service.update_category(db, user_id, category_id, data)
    return {"code": 0, "message": "更新成功", "data": item}


@router.delete("/{category_id}")
def delete_category(category_id: int, confirm: bool = Query(False), user_id: int = Depends(get_current_user_id), db: Session = Depends(get_db)):
    result = category_service.delete_category(db, user_id, category_id, confirm)
    return {"code": 0, "message": result["message"], "data": None}
